src/dataflow/src/sink/kafka.rs

```diff
@@ -574,3 +581,161 @@ where
 
     Box::new(KafkaSinkToken { shutdown_flag })
 }
+
+/// An operator for encoding Rows as Avro that can yield, and doesn't have to process all its in
+/// put in one invocation.
+fn avro_encode_stream<G>(
```

Comment:
I'm going to be a pain in the butt and ask for some more documentation here. Can you say what properties the input stream should have coming in, and what properties the output stream is guaranteed to have at its conclusion? There are nearby bits of code that want to assume things like operators don't shuffle dat, maybe maintain partitioning by key. This code may not have any thoughts there, or it might, but if you can jot them down as part of the method doccomments, that would avoid future confusion.

Reply:
Absolutely!